diff --git a/src/lib.rs b/src/lib.rs
index 7b69a87..32b90d8 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -190,11 +190,11 @@ macro_rules! make_struct {
             // XXX: The request ID is composed of 4 64-bit wide unsigned integers but the first
             //      two remain constant. This is refered as `_COMMON_MAGIC` in the  protocol
             //      header.
-            pub const ID: [u64; 4] = [0xc7b1dd30df4c8b88, 0x0a82e883a194f07b, $id1, $id2];
+            // pub const ID: [u64; 4] = [0xc7b1dd30df4c8b88, 0x0a82e883a194f07b, $id1, $id2];
 
             pub const fn new(revision: u64) -> Self {
                 Self {
-                    id: Self::ID,
+                    id: [0xc7b1dd30df4c8b88, 0x0a82e883a194f07b, $id1, $id2],
                     revision,
 
                     response: UnsafeCell::new(Ptr::DEFAULT),
